### A rental car company claims the mean time to rent a car on their website is 60 seconds with a standard deviation of 30 seconds. A random sample of 36 customers attempted to rent a car on the website. 

### 1) If the mean time to rent was 65 seconds. Is this enough evidence to contradict the company's claim at 5% confidence level? What's the power?

In [12]:
import scipy.stats as stats
import math

H0: μ = μ0          
H1: μ ≠ μ0

In [6]:
μ = 65
μ0 = 60
std = 30
n = 36
Z = (μ-μ0)/(std/math.sqrt(n))
print('Z_Score: ' + str(Z))

Z_Score: 1.0


In [11]:
power = round(stats.norm.cdf(Z), 3)
p_value =round(1 - power, 3)
print('p_value: ' + str(p_value))
print('power: ' + str(power*100) +'%')

p_value: 0.159
power: 84.1%


#### Because the p_value is bigger than the significance level at 5%, we cannot reject the null hyphothesis. 
#### Power is 84.1%.

### 2) If the analysts team believe the claim is not correct and the the time for rent is 67 seconds, what's the sample size need to designed for a sufficient test with 85% fo power

In [16]:
new_Z = stats.norm.ppf(0.85)
print('The new Z score: ' + str(new_Z))

The new Z score: 1.0364333894937898


In [21]:
new_μ = 67
new_n = ((new_Z*std)/(new_μ-μ0))**2
new_n

19.730097015751575